jwst/datamodels/schemas/core.schema.yaml

```diff
@@ -782,7 +782,7 @@ properties:
               # Obsolete NIRCam
               MASKA210R, MASKA430R, MASKBLWB, MASKBSWB,
               # NIRISS
-              SUB64, SUB80, SUB128, SUB256, SUBAMPCAL, SUBSTRIP96, SUBSTRIP256,
+              SUB80, SUBAMPCAL, SUBSTRIP96, SUBSTRIP256,
```

Comment:
Since these are defined as part of an `enum`, it's not valid to have duplicate values here. Ditto on the change below.

Reply:
Looks like NIRISS and MIRI have some of the same subarray names.  One of the disadvantages of having an `enum` that applies to all instruments, given the way we've organized it in the code here.  I suspect a `set()` of all these would create a valid enum, but I suspect we can't do that in YAML schemas.

Reply:
You could define a section with:
# MIRI, NIRISS
and list the common subarray names there.

Reply:
That's a good idea Phil.

Reply:
For CRDS I've tried to exploit this by-instrument breakdown and transcribe the enums to add extra checks to CRDS.   So,  where needed,  CRDS verifies (a) that SUBARRAY is definitely defined and (b) that it has instrument-specific values.   CRDS also uses these enums to check rmaps which can be directly edited by people.

Can I count on these by-instrument breakdowns to remain accurate?   Is there some other more definitive source than the CAL schema which breaks value enums down by instrument?

Reply:
The JWST Keyword Dictionary (json) schemas have keywords like FILTER, PUPIL, READPATT, SUBARRAY, etc. broken down by instrument.